scripts/manage-db.py

```diff
@@ -13,22 +14,77 @@
 
 from auslib.db import AUSDatabase
 
+
+def cleanup_releases(trans, nightly_age, dryrun=True):
+    # This and the subsequent queries use "%%%%%" because we end up going
+    # through two levels of Python string formatting. The first is here,
+    # and the second happens at a low level of SQLAlchemy when the transaction
+    # is being executed.
+    query = """
+LEFT JOIN releases_history USING (name)
+LEFT JOIN rules rules_mapping ON (name=rules_mapping.mapping)
+LEFT JOIN rules rules_whitelist ON (name=rules_whitelist.whitelist)
+WHERE name LIKE '%%%%nightly%%%%'
+AND (rules_whitelist.whitelist <> releases.name OR rules_whitelist.whitelist IS NULL)
+AND (rules_mapping.mapping <> releases.name OR rules_mapping.mapping IS NULL)
+AND (timestamp < (1000 * UNIX_TIMESTAMP(NOW() - INTERVAL %s MONTH)) OR change_id is NULL);
```

Comment:
Something about this still jangles the warning bell at the back of my head, so either I'm not getting it or something is amiss. Perhaps this ...

IIRC the `change_id IS NULL` condition would only be true if there is no history, say if `cleanup_releases_history()` removed history older than 7 days. So `%nightly%` are removed from releases if we have history going back X months, or after 7 days + the cron period ? Seems a bit incongruous, and the timestamp comparison is expensive if it never catches anything.

If we just want 7 day lifetime for (unmapped) nightlies and their history then we could remove the history first, and avoid the timestamp check here. There may be an edgecase around restored database without history, or maybe a new db with direct inserts (but don't do that!).

Or we may want to keep the release for longer than its history, which would be useful on the odd occasion we don't get informed of watersheds soon enough. The query could select releases which are unmapped & unwhitelisted, and the name contains a date older than some threshold. This could be fairly cheap by avoiding a join to the releases_history table.

Also, I don't grok the condition `rules_whitelist.whitelist <> releases.name`, that seems impossible given the join. Is it just to allow `rules_whitelist.whitelist IS NULL` as a check ? Can we write it as 
 `... AND (rules_mapping.mapping IS NULL AND rules_whitelist.whitelist IS NULL) ...` ?

The correct response to all this may be 'Nick should go learn some more SQL', or 'stop optimising this thing that won't take very long to run on RDS'; I'm just trying to figure out what the query is doing. If you have confidence in it from experimenting with live data and seeing the output then do squash my brainos and we can move along.